myapp/app/controllers/tasks_controller.rb

```diff
@@ -4,14 +4,20 @@
 
 # TaskController is a controller to handle basic CRUD operations for "task"
 class TasksController < ApplicationController
-  before_action :redirect_to_login_path_if_not_logged_in
+  before_action :redirect_to_login_path_if_not_logged_in, :redirect_to_maintenance
 
   def index
     q = Task
 
     unless params[:query].to_s.empty?
       # TODO: fix this. this fuzzy search might cause performance degradation.
       q = q.where('title LIKE ?', "%#{ActiveRecord::Base.sanitize_sql_like(params[:query].to_s)}%")
+
```

Comment:
In Ruby, you often use `Task` directly instead of `q = Task`.
```suggestion
tasks = Task.where('title LIKE ?', "%#{ActiveRecord::Base.sanitize_sql_like(params[:query].to_s)}%")
```

Reply:
Thanks for suggestion, just committed as you suggested.